packages/vite/src/node/config.ts

```diff
@@ -570,31 +570,54 @@ export async function resolveConfig(
   }
 
   // Check if all assetFileNames have the same reference.
-  // If not, display a warn for user.
+  // If not, display a warn for user. (exclude lib mode, it will make all asset to base64)
   const outputOption = config.build?.rollupOptions?.output ?? []
   // Use isArray to narrow its type to array
-  if (Array.isArray(outputOption)) {
-    const assetFileNamesList = outputOption.map(
-      (output) => output.assetFileNames
-    )
-    if (assetFileNamesList.length > 1) {
-      const firstAssetFileNames = assetFileNamesList[0]
-      const hasDifferentReference = assetFileNamesList.some(
-        (assetFileNames) => assetFileNames !== firstAssetFileNames
+  if (
+    !config.build?.lib &&
+    !verifyRollupOutputOption(outputOption, 'assetFileNames')
+  ) {
+    resolved.logger.warn(
+      colors.yellow(
+        `assetFileNames isn't equal for every build.rollupOptions.output. ` +
+          `A single pattern across all outputs is supported by Vite.`
       )
-      if (hasDifferentReference) {
-        resolved.logger.warn(
-          colors.yellow(`
-assetFileNames isn't equal for every build.rollupOptions.output. A single pattern across all outputs is supported by Vite.
-`)
-        )
-      }
-    }
+    )
   }
 
+  const workerOutputOption = config.worker?.rollupOptions?.output ?? []
+  const differentOpts = [
+    !verifyRollupOutputOption(workerOutputOption, 'assetFileNames') &&
+      'assetFileNames',
+    !verifyRollupOutputOption(workerOutputOption, 'chunkFileNames') &&
+      'chunkFileNames',
+    !verifyRollupOutputOption(workerOutputOption, 'entryFileNames') &&
+      'entryFileNames'
```

Comment:
If we need this, there is a space missing in the join below.

But could you explain why `chunkFileNames` and `entryFileNames` were added here? Are we also using the first one in case the output is an array somewhere?

Reply:
because worker bundle should emit all the assets(asset/chunk/entry) by hand. Only one difference needs warning

Reply:
Oh, I see. This only affects workers though, no? Maybe we should separate the warnings for them. So only getting these when using the workers? (cached for perf?)
It may be confusing if not for people not using the worker feature 🤔

Reply:
Ya, sounds good!